docs/resources/repository.md

```diff
@@ -141,9 +146,13 @@ Required:
 - `server_type` (String) Type of the MongoDB server. Allowed values:
   - `replicaset`
   - `standalone`
-  - `sharded`
+  - `sharded`.
+    The following conditions apply:
+  - If `server_type=standalone`, then only one `repo_node` block can be declared and it must be static. See `dynamic`. - If `server_type=sharded` and the a `srv_record_name` is provided, then all `repo_node` blocks must be declared dynamic, otherwise all of them must be static. See `dynamic`.
 
 Optional:
 
 - `replica_set_name` (String) Name of the replica set, if applicable.
-- `srv_record_name` (String) Name of a DNS SRV record which contains cluster topology details
+- `srv_record_name` (String) Name of a DNS SRV record which contains cluster topology details. The following conditions apply:
+  - If `srv_record_name` is specified, then all `repo_node` blocks can be declared `dynamic=true`.
+  - If `srv_record_name` is _not_ specified, then at _least one_ node must have `repo_node.host` and `repo_node.port`.
```

Comment:
I feel like here we're kinda giving them mixed instructions. Here we say:

> If `srv_record_name` is _not_ specified, then at _least one_ node must have `repo_node.host` and `repo_node.port`.

This means that I can mix static and dynamic, right? However, above, we have:

> If `server_type=sharded` and the a `srv_record_name` is provided, then all `repo_node` blocks must be declared dynamic, otherwise all of them must be static. See `dynamic`

This means that we *cannot* mix.

So, if people read the former, they might be tempted to mix stuff (which is wrong, and which will result in the API returning an error), and if they read the latter, they will do the right thing. If they read both... then they will be confused :stuck_out_tongue_winking_eye: 

Maybe we should repeat the part about not mixing static and dynamic in the former?

Reply:
> This means that I can mix static and dynamic, right?
Yes, this is what I understood from this internal statement: `at least one node must be static (not dynamic) UNLESS SRV record is specified in mongodb settings`, which means that we should be able to mix dynamic and static if the SRV record is not present and as long as the first node is static.

Now I am looking at [this code](https://github.com/cyralinc/crud/blob/main/repo/repo_crud.go#L125-L156), I see that if the `srv_record_name` is **not** specified, we check:
- If cluster type is `sharded`, then all nodes must be static;
- If cluster type is `standalone`,  then a single node must be declared and it needs to be static (following [this check](https://github.com/cyralinc/crud/blob/main/repo/repo_crud.go#L155) and [this one](https://github.com/cyralinc/crud/blob/22ba2cb912d6f8cee98d2970e586d558776587f6/repo/repo_crud.go#L1549)).
- As `replicaset` is not ruled out from that code, [this line](https://github.com/cyralinc/crud/blob/main/repo/repo_crud.go#L155) allows us to have mixed configurations of static+dynamic as long as the first node is static.

By contrast, if `srv_record_name` is specified, we check [here](https://github.com/cyralinc/crud/blob/22ba2cb912d6f8cee98d2970e586d558776587f6/repo/repo_crud.go#L158-L167):
- All nodes must be dynamic.


I will update the docs accordingly to those rules. LMK if I missed something.

Reply:
Given the rules above, we could simplify the documentation and move it to the proper attributes as follows:

`srv_record_name`
- If provided, then all nodes must be dynamic .

`server_type`
- If `sharded` and `srv_record_name` not provided, then all nodes must be static.
- If `standalone` and `srv_record_name` not provided, then only a single static node must be declared.
- If `replicaset` and `srv_record_name` not provided, then mixed configuration of static and dynamic nodes are allowed as long as the first node is static.